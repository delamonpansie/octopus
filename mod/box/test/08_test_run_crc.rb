#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

RunEnv.connect_eval do |env|
  ping
  33.times do |i|
    insert [i, i + 1, i + 2]
  end

  env.stop

  puts `OCTOPUS_CAT_RUN_CRC=1 ./octopus --cat 00000000000000000002.xlog 2>/dev/null | sed 's/tm:[^ ]* //'` + "\n"
end
