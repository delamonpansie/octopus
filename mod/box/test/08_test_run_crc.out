# box.ping()
:pong

# box.insert([0, 1, 2])
1

# box.insert([1, 2, 3])
1

# box.insert([2, 3, 4])
1

# box.insert([3, 4, 5])
1

# box.insert([4, 5, 6])
1

# box.insert([5, 6, 7])
1

# box.insert([6, 7, 8])
1

# box.insert([7, 8, 9])
1

# box.insert([8, 9, 10])
1

# box.insert([9, 10, 11])
1

# box.insert([10, 11, 12])
1

# box.insert([11, 12, 13])
1

# box.insert([12, 13, 14])
1

# box.insert([13, 14, 15])
1

# box.insert([14, 15, 16])
1

# box.insert([15, 16, 17])
1

# box.insert([16, 17, 18])
1

# box.insert([17, 18, 19])
1

# box.insert([18, 19, 20])
1

# box.insert([19, 20, 21])
1

# box.insert([20, 21, 22])
1

# box.insert([21, 22, 23])
1

# box.insert([22, 23, 24])
1

# box.insert([23, 24, 25])
1

# box.insert([24, 25, 26])
1

# box.insert([25, 26, 27])
1

# box.insert([26, 27, 28])
1

# box.insert([27, 28, 29])
1

# box.insert([28, 29, 30])
1

# box.insert([29, 30, 31])
1

# box.insert([30, 31, 32])
1

# box.insert([31, 32, 33])
1

# box.insert([32, 33, 34])
1

lsn:2 shard:0 run_crc:0xdd071fd3 scn:2 t:wal/usr13 INSERT n:0 flags:00000000 <0:"\x00\x00\x00\x00", 1:"\x01\x00\x00\x00", 2:"\x02\x00\x00\x00">
lsn:3 shard:0 run_crc:0x8b08ab74 scn:3 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00", 2:"\x02\x00\x00\x00", 3:"\x03\x00\x00\x00">
lsn:4 shard:0 run_crc:0x97aacb8c scn:4 t:wal/usr13 INSERT n:0 flags:00000000 <2:"\x02\x00\x00\x00", 3:"\x03\x00\x00\x00", 4:"\x04\x00\x00\x00">
lsn:5 shard:0 run_crc:0xf7cc721e scn:5 t:wal/usr13 INSERT n:0 flags:00000000 <3:"\x03\x00\x00\x00", 4:"\x04\x00\x00\x00", 5:"\x05\x00\x00\x00">
lsn:6 shard:0 run_crc:0x387209b8 scn:6 t:wal/usr13 INSERT n:0 flags:00000000 <4:"\x04\x00\x00\x00", 5:"\x05\x00\x00\x00", 6:"\x06\x00\x00\x00">
lsn:7 shard:0 run_crc:0x069205e7 scn:7 t:wal/usr13 INSERT n:0 flags:00000000 <5:"\x05\x00\x00\x00", 6:"\x06\x00\x00\x00", 7:"\x07\x00\x00\x00">
lsn:8 shard:0 run_crc:0x9ec8617a scn:8 t:wal/usr13 INSERT n:0 flags:00000000 <6:"\x06\x00\x00\x00", 7:"\x07\x00\x00\x00", 8:"\x08\x00\x00\x00">
lsn:9 shard:0 run_crc:0x9c7afe2a scn:9 t:wal/usr13 INSERT n:0 flags:00000000 <7:"\x07\x00\x00\x00", 8:"\x08\x00\x00\x00", 9:"\x09\x00\x00\x00">
lsn:10 shard:0 run_crc:0x914e0c6f scn:10 t:wal/usr13 INSERT n:0 flags:00000000 <8:"\x08\x00\x00\x00", 9:"\x09\x00\x00\x00", 10:"\x0A\x00\x00\x00">
lsn:11 shard:0 run_crc:0x049f1658 scn:11 t:wal/usr13 INSERT n:0 flags:00000000 <9:"\x09\x00\x00\x00", 10:"\x0A\x00\x00\x00", 11:"\x0B\x00\x00\x00">
lsn:12 shard:0 run_crc:0xaf316caa scn:12 t:wal/usr13 INSERT n:0 flags:00000000 <10:"\x0A\x00\x00\x00", 11:"\x0B\x00\x00\x00", 12:"\x0C\x00\x00\x00">
lsn:13 shard:0 run_crc:0x3f492bab scn:13 t:wal/usr13 INSERT n:0 flags:00000000 <11:"\x0B\x00\x00\x00", 12:"\x0C\x00\x00\x00", 13:"\x0D\x00\x00\x00">
lsn:14 shard:0 run_crc:0x8f111e2d scn:14 t:wal/usr13 INSERT n:0 flags:00000000 <12:"\x0C\x00\x00\x00", 13:"\x0D\x00\x00\x00", 14:"\x0E\x00\x00\x00">
lsn:15 shard:0 run_crc:0x265cafd0 scn:15 t:wal/usr13 INSERT n:0 flags:00000000 <13:"\x0D\x00\x00\x00", 14:"\x0E\x00\x00\x00", 15:"\x0F\x00\x00\x00">
lsn:16 shard:0 run_crc:0x74b199e3 scn:16 t:wal/usr13 INSERT n:0 flags:00000000 <14:"\x0E\x00\x00\x00", 15:"\x0F\x00\x00\x00", 16:"\x10\x00\x00\x00">
lsn:17 shard:0 run_crc:0xb7442885 scn:17 t:wal/usr13 INSERT n:0 flags:00000000 <15:"\x0F\x00\x00\x00", 16:"\x10\x00\x00\x00", 17:"\x11\x00\x00\x00">
lsn:18 shard:0 run_crc:0xe7c9d8e0 scn:18 t:wal/usr13 INSERT n:0 flags:00000000 <16:"\x10\x00\x00\x00", 17:"\x11\x00\x00\x00", 18:"\x12\x00\x00\x00">
lsn:19 shard:0 run_crc:0xf49d5c52 scn:19 t:wal/usr13 INSERT n:0 flags:00000000 <17:"\x11\x00\x00\x00", 18:"\x12\x00\x00\x00", 19:"\x13\x00\x00\x00">
lsn:20 shard:0 run_crc:0x4b24a484 scn:20 t:wal/usr13 INSERT n:0 flags:00000000 <18:"\x12\x00\x00\x00", 19:"\x13\x00\x00\x00", 20:"\x14\x00\x00\x00">
lsn:21 shard:0 run_crc:0xcb283487 scn:21 t:wal/usr13 INSERT n:0 flags:00000000 <19:"\x13\x00\x00\x00", 20:"\x14\x00\x00\x00", 21:"\x15\x00\x00\x00">
lsn:22 shard:0 run_crc:0xa4230fe0 scn:22 t:wal/usr13 INSERT n:0 flags:00000000 <20:"\x14\x00\x00\x00", 21:"\x15\x00\x00\x00", 22:"\x16\x00\x00\x00">
lsn:23 shard:0 run_crc:0x110f48ee scn:23 t:wal/usr13 INSERT n:0 flags:00000000 <21:"\x15\x00\x00\x00", 22:"\x16\x00\x00\x00", 23:"\x17\x00\x00\x00">
lsn:24 shard:0 run_crc:0xb6891529 scn:24 t:wal/usr13 INSERT n:0 flags:00000000 <22:"\x16\x00\x00\x00", 23:"\x17\x00\x00\x00", 24:"\x18\x00\x00\x00">
lsn:25 shard:0 run_crc:0x4f15eb2b scn:25 t:wal/usr13 INSERT n:0 flags:00000000 <23:"\x17\x00\x00\x00", 24:"\x18\x00\x00\x00", 25:"\x19\x00\x00\x00">
lsn:26 shard:0 run_crc:0xe47c00d7 scn:26 t:wal/usr13 INSERT n:0 flags:00000000 <24:"\x18\x00\x00\x00", 25:"\x19\x00\x00\x00", 26:"\x1A\x00\x00\x00">
lsn:27 shard:0 run_crc:0xc332cbd2 scn:27 t:wal/usr13 INSERT n:0 flags:00000000 <25:"\x19\x00\x00\x00", 26:"\x1A\x00\x00\x00", 27:"\x1B\x00\x00\x00">
lsn:28 shard:0 run_crc:0x2ba8af0c scn:28 t:wal/usr13 INSERT n:0 flags:00000000 <26:"\x1A\x00\x00\x00", 27:"\x1B\x00\x00\x00", 28:"\x1C\x00\x00\x00">
lsn:29 shard:0 run_crc:0xdceb5e02 scn:29 t:wal/usr13 INSERT n:0 flags:00000000 <27:"\x1B\x00\x00\x00", 28:"\x1C\x00\x00\x00", 29:"\x1D\x00\x00\x00">
lsn:30 shard:0 run_crc:0x072ef70d scn:30 t:wal/usr13 INSERT n:0 flags:00000000 <28:"\x1C\x00\x00\x00", 29:"\x1D\x00\x00\x00", 30:"\x1E\x00\x00\x00">
lsn:31 shard:0 run_crc:0x4b611076 scn:31 t:wal/usr13 INSERT n:0 flags:00000000 <29:"\x1D\x00\x00\x00", 30:"\x1E\x00\x00\x00", 31:"\x1F\x00\x00\x00">
lsn:32 shard:0 run_crc:0xc0240090 scn:32 t:wal/usr13 INSERT n:0 flags:00000000 <30:"\x1E\x00\x00\x00", 31:"\x1F\x00\x00\x00", 32:" \x00\x00\x00">
lsn:33 shard:0 run_crc:0x08041b8a scn:33 t:wal/usr13 INSERT n:0 flags:00000000 <31:"\x1F\x00\x00\x00", 32:" \x00\x00\x00", 33:"!\x00\x00\x00">
lsn:34 shard:0 run_crc:0x21bf1a2a scn:34 t:wal/usr13 INSERT n:0 flags:00000000 <32:" \x00\x00\x00", 33:"!\x00\x00\x00", 34:"\x22\x00\x00\x00">

